crossopt/assignment1/Q2.py

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env python3
+
+
+class Node:
+    def __init__(self, data):
+        self.data = data
+        self.next = None
+
+
+class LinkedList:
+    def __init__(self):
+        self.head = None
+
+    # adds element to front of list
+    def insert(self, new_elem):
+        new_node = Node(new_elem)
+        new_node.next = self.head
+        self.head = new_node
+
+    def length(self):
```

Comment:
I see below that you know about the __str__() that works with print.
Here in the same vein, you can use __len__() instead of length() to have the Python builtin len() function that still works.